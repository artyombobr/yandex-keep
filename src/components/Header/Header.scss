@import "../../shared";

.header {
  position: fixed;
  top: 0;
  z-index: 3;
  width: 100%;
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  background-color: #fff;

  &__logo {
    height: 64px;
    margin: 0 16px;
  }

  &__checkbox {
    display: none;
  }

  &__hamburger {
    position: absolute;
    top: 31px;
    right: 17px;
  }

  &__right {
    display: flex;
    margin: 0 16px;
    flex-direction: inherit;
    max-height: 0;
    overflow: hidden;
  }
}

.menu {
  display: flex;
  flex-direction: inherit;
  margin-bottom: 16px;

  &__link{
    font-size: 16px;
    line-height: 32px;
    margin-bottom: 16px;
    text-align: center;
    text-decoration: none;
    transition: color .4s;

    &_gray {
      color: rgba(51, 51, 51, .6);
    }

    &:hover {
      color: #000;
    }
  }

  &__button{
    font-size: 15px;
    line-height: 32px;
    height: 32px;
    background: #FC0;
    border-radius: 4px;
    border: none;
    outline: none;
    cursor: pointer;
    transition: background .4s;

    &:hover {
      background: #ff9d00;
    }
  }
}

.right__search{
  height: 32px;
  margin-bottom: 16px;
}

.header ul {
  display: flex;
  flex-direction: inherit;
}

.navicon {
  background: #333;
  display: block;
  height: 2px;
  position: relative;
  transition: background .2s ease-out;
  width: 22px;

  &::before,
  &::after {
    background: #333;
    content: '';
    display: block;
    height: 100%;
    position: absolute;
    transition: all .2s ease-out;
    width: 100%;
  }

  &::before {
    top: -7px;
  }

  &::after {
    top: 7px;
  }
}

.header__checkbox:checked ~ .header__right {
  max-height: 240px;
}

@media (min-width: $medium) {
  .header {
    flex-direction: row;
    align-items: center;

    &__menu {
      flex-direction: row;
      margin-bottom: 0;
      margin-left: 32px;
    }

    &__hamburger {
      display: none;
    }

    &__right {
      flex-direction: row-reverse;
      flex-grow: 1;
      max-height: 240px;
    }
  }

  .search{
    padding-right: calc(50% - 467px);
    max-width: 474px;
    margin-bottom: 0;
  }

  .menu {
    flex-direction: row;
    margin-left: 20px;
    margin-bottom: 0;

    &__link {
      margin-left: 32px;
      margin-bottom: 0;
    }

    &__button {
      margin-left: 32px;
      padding: 0 15px;
    }
  }
}
