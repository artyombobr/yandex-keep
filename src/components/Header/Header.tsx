import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../Logo/Logo';
import Search from '../Search/Search';
import './Header.scss';

const Header = () => (
  <header className="header">
    <Logo className="header__logo" color="#000" />
    <input className="header__checkbox" type="checkbox" id="toggle" />
    <label className="header__hamburger" htmlFor="toggle">
      <span className="navicon" />
    </label>
    <div className="header__right right">
      <div className="right__menu menu">
        <Link to="/">
          <button className="menu__link" type="button">
            {'Активные'}
          </button>
        </Link>
        <Link to="/archive">
          <button className="menu__link menu__link_gray" type="button">
            {'Архив'}
          </button>
        </Link>
        <button type="button" className="menu__button">
          {'Добавить'}
        </button>
      </div>
      <Search className="right__search" />
    </div>
  </header>
);

export default Header;
