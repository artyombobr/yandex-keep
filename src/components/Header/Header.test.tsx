import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header', () => {
  it('correct render', () => {
    const component = shallow(
      <Header isVisibleModal={false} dispatch={() => {}} />
    );
    expect(component).toMatchSnapshot();
  });
});
