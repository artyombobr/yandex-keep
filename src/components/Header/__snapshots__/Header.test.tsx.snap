// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    dispatch={[Function]}
    isVisibleModal={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Logo
          className="header__logo"
          color="#000"
        />,
        <input
          className="header__checkbox"
          id="toggle"
          type="checkbox"
        />,
        <label
          className="header__hamburger"
          htmlFor="toggle"
        >
          <span
            className="navicon"
          />
        </label>,
        <div
          className="header__right right"
        >
          <div
            className="right__menu menu"
          >
            <NavLink
              activeClassName="menu__link_active"
              className="menu__link"
              exact={true}
              to="/"
            >
              Активные
            </NavLink>
            <NavLink
              activeClassName="menu__link_active"
              className="menu__link"
              to="/archive"
            >
              Архив
            </NavLink>
            <button
              className="menu__button"
              onClick={[Function]}
              type="button"
            >
              Добавить
            </button>
          </div>
          <UNDEFINED
            className="right__search"
          />
        </div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "header__logo",
          "color": "#000",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "header__checkbox",
          "id": "toggle",
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="navicon"
          />,
          "className": "header__hamburger",
          "htmlFor": "toggle",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "navicon",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="right__menu menu"
            >
              <NavLink
                activeClassName="menu__link_active"
                className="menu__link"
                exact={true}
                to="/"
              >
                Активные
              </NavLink>
              <NavLink
                activeClassName="menu__link_active"
                className="menu__link"
                to="/archive"
              >
                Архив
              </NavLink>
              <button
                className="menu__button"
                onClick={[Function]}
                type="button"
              >
                Добавить
              </button>
            </div>,
            <UNDEFINED
              className="right__search"
            />,
          ],
          "className": "header__right right",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <NavLink
                  activeClassName="menu__link_active"
                  className="menu__link"
                  exact={true}
                  to="/"
                >
                  Активные
                </NavLink>,
                <NavLink
                  activeClassName="menu__link_active"
                  className="menu__link"
                  to="/archive"
                >
                  Архив
                </NavLink>,
                <button
                  className="menu__button"
                  onClick={[Function]}
                  type="button"
                >
                  Добавить
                </button>,
              ],
              "className": "right__menu menu",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "menu__link_active",
                  "children": "Активные",
                  "className": "menu__link",
                  "exact": true,
                  "to": "/",
                },
                "ref": null,
                "rendered": "Активные",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "menu__link_active",
                  "children": "Архив",
                  "className": "menu__link",
                  "to": "/archive",
                },
                "ref": null,
                "rendered": "Архив",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Добавить",
                  "className": "menu__button",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Добавить",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "right__search",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Search)",
              "type": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Logo
            className="header__logo"
            color="#000"
          />,
          <input
            className="header__checkbox"
            id="toggle"
            type="checkbox"
          />,
          <label
            className="header__hamburger"
            htmlFor="toggle"
          >
            <span
              className="navicon"
            />
          </label>,
          <div
            className="header__right right"
          >
            <div
              className="right__menu menu"
            >
              <NavLink
                activeClassName="menu__link_active"
                className="menu__link"
                exact={true}
                to="/"
              >
                Активные
              </NavLink>
              <NavLink
                activeClassName="menu__link_active"
                className="menu__link"
                to="/archive"
              >
                Архив
              </NavLink>
              <button
                className="menu__button"
                onClick={[Function]}
                type="button"
              >
                Добавить
              </button>
            </div>
            <UNDEFINED
              className="right__search"
            />
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "header__logo",
            "color": "#000",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "header__checkbox",
            "id": "toggle",
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span
              className="navicon"
            />,
            "className": "header__hamburger",
            "htmlFor": "toggle",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "navicon",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="right__menu menu"
              >
                <NavLink
                  activeClassName="menu__link_active"
                  className="menu__link"
                  exact={true}
                  to="/"
                >
                  Активные
                </NavLink>
                <NavLink
                  activeClassName="menu__link_active"
                  className="menu__link"
                  to="/archive"
                >
                  Архив
                </NavLink>
                <button
                  className="menu__button"
                  onClick={[Function]}
                  type="button"
                >
                  Добавить
                </button>
              </div>,
              <UNDEFINED
                className="right__search"
              />,
            ],
            "className": "header__right right",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <NavLink
                    activeClassName="menu__link_active"
                    className="menu__link"
                    exact={true}
                    to="/"
                  >
                    Активные
                  </NavLink>,
                  <NavLink
                    activeClassName="menu__link_active"
                    className="menu__link"
                    to="/archive"
                  >
                    Архив
                  </NavLink>,
                  <button
                    className="menu__button"
                    onClick={[Function]}
                    type="button"
                  >
                    Добавить
                  </button>,
                ],
                "className": "right__menu menu",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "menu__link_active",
                    "children": "Активные",
                    "className": "menu__link",
                    "exact": true,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "Активные",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "activeClassName": "menu__link_active",
                    "children": "Архив",
                    "className": "menu__link",
                    "to": "/archive",
                  },
                  "ref": null,
                  "rendered": "Архив",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Добавить",
                    "className": "menu__button",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "Добавить",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "right__search",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.memo),
                "WrappedComponent": [Function],
                "compare": null,
                "displayName": "Connect(Search)",
                "type": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
